xbmc/filesystem/Directory.h

```diff
@@ -9,6 +9,7 @@
 #pragma once
 
 #include "IDirectory.h"
+#include "addons/addoninfo/AddonType.h"
```

Comment:
This doesn't look right. Plugins are just a type of vfs directory while the directory include is common vfs code

Reply:
I agree, this is an api leak. However, as this bug shows, the assumption that is made in the api that directory code doesn't need information from the addon is incorrect. I made a change to pass this information without leaking any implementation in the api surface. Please take a look at the latest push.